optaweb-employee-rostering-backend/src/main/java/org/optaweb/employeerostering/service/roster/RosterGenerator.java

```diff
@@ -493,20 +503,43 @@ public RosterState createRosterState(GeneratorType generatorType, Tenant tenant,
                                             timeslotRangesIndex + ").");
                         }
                         // There might be less employees than we need (overconstrained planning)
-                        Employee rotationEmployee = rotationEmployeeIndex >= rotationEmployeeList.size() ? null :
-                                rotationEmployeeList.get(rotationEmployeeIndex);
+                        Employee rotationEmployee = rotationEmployeeIndex >= doctorRotationEmployeeList.size() ? null :
+                                doctorRotationEmployeeList.get(rotationEmployeeIndex);
                         ShiftTemplate shiftTemplate = new ShiftTemplate(tenantId, spot, startDayOffset, startTime,
-                                                                        endDayOffset, endTime, rotationEmployee);
+                                                                        endDayOffset, endTime, rotationEmployee,
+                                                                        new HashSet<>(Arrays.asList(skillList.get(1))));
+                        entityManager.persist(shiftTemplate);
+                        shiftTemplateList.add(shiftTemplate);
+
+                        rotationEmployeeIndex = generatorType.rotationEmployeeIndexCalculator
+                                .apply(startDayOffset, timeslotRangesIndex).get(0);
+                        rotationEmployee = rotationEmployeeIndex >= nurseRotationEmployeeList.size() ? null :
+                                nurseRotationEmployeeList.get(rotationEmployeeIndex);
+                        shiftTemplate = new ShiftTemplate(tenantId, spot, startDayOffset, startTime,
+                                                          endDayOffset, endTime, rotationEmployee,
+                                                          new HashSet<>(Arrays.asList(skillList.get(2))));
+                        entityManager.persist(shiftTemplate);
+                        shiftTemplateList.add(shiftTemplate);
+
+                        rotationEmployeeIndex = generatorType.rotationEmployeeIndexCalculator
+                                .apply(startDayOffset, timeslotRangesIndex).get(1);
+                        rotationEmployee = rotationEmployeeIndex >= nurseRotationEmployeeList.size() ? null :
+                                nurseRotationEmployeeList.get(rotationEmployeeIndex);
+                        shiftTemplate = new ShiftTemplate(tenantId, spot, startDayOffset, startTime,
+                                                          endDayOffset, endTime, rotationEmployee,
+                                                          new HashSet<>(Arrays.asList(skillList.get(2))));
```

Comment:
I disagree using varargs, because then you cannot pass a Collection directly in. I think  a good compromise would be making it take a Collection, and create a new HashSet in the constructor. WDYT?